src/Akeneo/Bundle/BatchBundle/Command/BatchCommand.php

```
@@ -78,6 +85,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $logger->pushHandler(new StreamHandler('php://stdout'));
         }
 
+        $username = $input->getOption('username');
+
+        try {
+            $user = $this->getContainer()->get('security.user.provider.concrete.chain_provider')
+                ->loadUserByUsername($username);
+
+            $token = new UsernamePasswordToken($user, null, 'main', $user->getRoles());
+            $this->getContainer()->get('security.token_storage')->setToken($token);
+        } catch (UsernameNotFoundException $e) {
+            // We do nothing as we want to let the possibility to not pass an user.
```

If `username` is optional, you should embed your ``try/catch`` inside an if condition: ``if (!is_null($username)) {}``.

But as a developer, if the username I use is wrong (I did a typo for instance) I want to be notified, so please, dont hide the error! At least, if you disagree with me, can you log or output something? 

```php
$warning = sprintf('<warning>User with username %s was not found</warning>', $username);
$output->writeln($warning);
```